### PR TITLE
pdal: new package

### PR DESCRIPTION
**Pull request by Sinan81**
new package pdal

### CODE REVIEW

**Comment by Sinan81** [comment]
pinging @adamjstewart for review if available.

**Comment by Sinan81** [comment]
@spackbot fix style

**Review by alecbcs — CHANGES_REQUESTED** [review]
@Sinan81 can you include the license of PDAL using the new `license()` directive?

File: var/spack/repos/builtin/packages/pdal/package.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class Pdal(CMakePackage):
+    """PDAL is a C++ library for translating and manipulating point cloud data.
+    It is very much like the GDAL library which handles raster and vector data.
+    """
+
+    license("BSD Open Source license.")
+
+    homepage = "https://pdal.io"
+    url = "https://github.com/PDAL/PDAL/archive/refs/tags/2.6.2.tar.gz"
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/pdal/package.py]:
```suggestion
    homepage = "https://pdal.io"
    url = "https://github.com/PDAL/PDAL/archive/refs/tags/2.6.2.tar.gz"
    
    license("BSD Open Source license.")
```

**Review by alecbcs — CHANGES_REQUESTED** [review]
Super small nitpick, `license()` is placed after the homepage and url variables and before `version()` directives. Otherwise this PR looks good to me!

**Comment by alecbcs** [comment]
@spackbot fix formatting

**Comment by alecbcs** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/pdal/package.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class Pdal(CMakePackage):
+    """PDAL is a C++ library for translating and manipulating point cloud data.
+    It is very much like the GDAL library which handles raster and vector data.
+    """
+
+    homepage = "https://pdal.io"
+    url = "https://github.com/PDAL/PDAL/archive/refs/tags/2.6.2.tar.gz"
+
+    license("BSD Open Source license.")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/pdal/package.py]:
```suggestion
    license("BSD")
```

Sorry I didn't catch this earlier, licenses should be specified in SPDX format as per the [docs](https://spack.readthedocs.io/en/latest/packaging_guide.html#specifying-license-information).

**Review by alecbcs — CHANGES_REQUESTED** [review]
Quick fix. Once that's in it looks good to me.

File: var/spack/repos/builtin/packages/pdal/package.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    depends_on("libgeotiff")
+    depends_on("proj@4.9.3:")
+
+    # https://github.com/PDAL/PDAL/issues/3826
+    patch("stdcppfs.patch", when="@:2.6.1 %gcc@:8")
+
+    def cmake_args(self):
+        args = []
+        args.append("-DPROJ_INCLUDE_DIR=" + self.spec["proj"].prefix.include)
+        return args
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/pdal/package.py]:
```suggestion
        return [
            self.define("PROJ_INCLUDE_DIR", self.spec["proj"].prefix.include),
        ]
```

We can use `self.define` here to make the syntax a bit nicer.

File: var/spack/repos/builtin/packages/pdal/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: I googled it. it seems there are multiple BSD versions. So, my understanding is that BSD open source license is different from the original BSD license.

File: var/spack/repos/builtin/packages/pdal/package.py (hunk 3 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: good to know.

File: var/spack/repos/builtin/packages/pdal/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: image

Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: let's get help from the developers of PDAL.

Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: https://github.com/PDAL/PDAL/blob/master/LICENSE.txt

Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: I'll approve the commit just in case.

Comment (Sinan81) [reply var/spack/repos/builtin/packages/pdal/package.py]: I'm told in the linked issue that the precise license is BSD-3-Clause-Attribution.html

**Comment by adamjstewart** [comment]
Can you list me as a maintainer of this recipe? @hobu  may be interested too.

**Comment by Sinan81** [comment]
@spackbot fix style

**Comment by hobu** [comment]
No thanks on being a maintainer. I already maintain the stack on conda forge and that is plenty.

File: var/spack/repos/builtin/packages/pdal/package.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    license("BSD")
+
+    version("2.6.2", sha256="ec4175cfe19dc6b70a0434850f837317f7202f84b63cd8dcc65ca83e04678f57")
+    version("2.6.1", sha256="da6e615f01b6110414ef3e2250f112e49df129091abc91ba6866bb01dc68454e")
+    version("2.6.0", sha256="12eedeac16ec3aaef42f06078f03f657701c25781207a8e09a3547519228780e")
+    version("2.5.6", sha256="de4177305e380d21187da8ec90afda64756bbde5e925035bd53e54a6e349df18")
+    version("2.5.5", sha256="6bf4f34bc0bf1bce52b8daecb03a7f45d218c0374bfa00783c787b9e54d56d72")
+    version("2.4.3", sha256="e1a910d593311e68b51f32d1f4f8fe4327b97ae7a8de209147b6111091b6f75b")
+    version("2.3.0", sha256="8ae848e9b3fe5149a9277fe60e10b9858edb9a3cf1a40728f11712498e5da13a")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pdal/package.py:28]: Build dep on cmake 3.13+

**Review by adamjstewart — APPROVED** [review]

**Review by alecbcs — APPROVED** [review]
Looks good to me.

**Pull request merged**